config.go

```diff
@@ -86,15 +83,23 @@ func (c *srcfileConfig) apply() error {
 		buildContext.GOROOT = cwd
 	}
 
-	if config.GOPATH != "" {
-		// clean/absolutize all paths
-		dirs := cleanDirs(filepath.SplitList(config.GOPATH))
-		config.GOPATH = strings.Join(dirs, string(filepath.ListSeparator))
-
-		dirs = append(dirs, filepath.SplitList(buildContext.GOPATH)...)
-		buildContext.GOPATH = strings.Join(uniq(dirs), string(filepath.ListSeparator))
-		loaderConfig.Build = &buildContext
+	// Automatically detect vendored dirs (check for vendor/src and
+	// Godeps/_workspace/src) and set up GOPATH pointing to them if
+	// they exist.
+	//
+	// Note that the `vendor` directory here is used by 3rd party vendoring
+	// tools and is NOT the `vendor` directory in the Go 1.6 official vendor
+	// specification (that `vendor` directory does not have a `src`
+	// subdirectory).
+	var gopaths []string
+	for _, vdir := range []string{"vendor", "Godeps/_workspace"} {
+		if fi, err := os.Stat(filepath.Join(cwd, vdir, "src")); err == nil && fi.Mode().IsDir() {
```

Comment:
Yes, afaik `cwd` is always absolute, because if it would be relative, then relative to what? The cwd? InfiniteRecursionException